lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -2523,11 +2535,16 @@ bool TypeChecker::diagnoseIfDeprecated(SourceLoc loc,
   // We match the behavior of clang to not report deprecation warnings
   // inside declarations that are themselves deprecated on all deployment
   // targets.
-  if (where.isDeprecated()) {
+  if (where.isDeprecated() || where.ignoresDeprecation()) {
     return false;
   }
 
   auto *dc = where.getDeclContext();
+  auto moduleImportedFrom = dc->findImportFor(dc->getParentModule());
+  if (moduleImportedFrom.hasValue() && moduleImportedFrom.getValue().options.contains(ImportFlags::IgnoresDeprecationWarnings)) {
+    return false;
+  }
+
```

Comment:
Same comment regarding the module you're querying, the name of the variable, and the syntactic sugar. In this case, the module you want is the parent module of the ExtensionDecl. Could you also add some test cases to make sure the attribute works for this case where an extension that declares a protocol conformance is marked as deprecated?

Reply:
As in,
```swift
@available(*, deprecated)
extension someStruct: aProtocol {}
```
?

Reply:
Yeah

Reply:
Is [this](https://github.com/apple/swift/pull/59232/commits/888f7aa038f8ec66e4fe212b9891c36356ca3703) good enough for this purpose?

Reply:
Sure, that looks great! Could you also add a case without `ignoreDeprecationWarnings` just to verify that it's the attribute silencing the warning?

Reply:
Sure